pkg/skaffold/deploy/resource/deployment.go

```diff
@@ -265,18 +286,21 @@ func (d *Deployment) fetchPods(ctx context.Context) error {
 	return nil
 }
 
-// Return first pod status in error.
-// TODO: should we return all distinct error codes in future?
-func (d *Deployment) FirstPodErrOccurred() proto.StatusCode {
-	if len(d.pods) == 0 {
-		return d.Status().ActionableError().ErrCode
+// StatusCode() returns the deployment status code if the status check is cancelled
+// or if no pod data exists for this deployment.
+// If pods are fetched, this function returns the error code a pod container encountered.
+func (d *Deployment) StatusCode() proto.StatusCode {
+	// do not process pod status codes if another deployment failed
+	// or the user aborted the run.
+	if d.statusCode == proto.StatusCode_STATUSCHECK_CONTEXT_CANCELLED {
```

Comment:
It feels really weird that we're handling a Go-level construct as an application deployment status code.

Reply:
`proto.StatusCode_STATUSCHECK_CONTEXT_CANCELLED` indicates, we could not complete the status check for a resource because user cancelled it or it was aborted.
I stayed away from "Aborted" to be specific regarding if it was aborted by user or because skaffold crashed.
Though this is internal detail, we should be capture this to exclude failure due to user actions. 
Does that make sense?
We can also rename this to "STATUSCHECK_USER_CANCELLED",  however i will have to see if this is already consumed by IDEs

Reply:
USER_CANCELLED is much better!

Reply:
yes. but as per our guidelines, we need the phase name.I can change it to`STATUSCHECK_USER_CANCELLED`

Reply:
done!